gnovm/pkg/gnolang/nodes.go

```diff
@@ -193,28 +193,62 @@ func (attr *Attributes) SetLabel(label Name) {
 }
 
 func (attr *Attributes) HasAttribute(key GnoAttribute) bool {
-	_, ok := attr.data[key]
+	_, _, ok := attr.getAttribute(key)
 	return ok
 }
 
 // GnoAttribute must not be user provided / arbitrary,
 // otherwise will create potential exploits.
 func (attr *Attributes) GetAttribute(key GnoAttribute) interface{} {
-	return attr.data[key]
+	val, _, _ := attr.getAttribute(key)
+	return val
+}
+
+func (attr *Attributes) getAttribute(key GnoAttribute) (any, int, bool) {
+	for i, kv := range attr.data {
+		if kv.key == key {
+			return kv.value, i, true
+		}
+	}
```

Comment:
If we want to avoid allocs here, wouldn't we want to not copy on range and instead use a for loop with iterator?
```suggestion
	for i := 0; i < len(attr.data); i++ {
		if attr.data[i].key == key {
			return attr.data[i].value, i, true
		}
	}
```